service/mongodb/controller.go

```diff
@@ -0,0 +1,114 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/meateam/fav-service/service"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+)
+
+// Controller is the favorite service business logic implementation using MongoStore.
+type Controller struct {
+	store MongoStore
+}
+
+// NewMongoController returns a new controller.
+func NewMongoController(db *mongo.Database) (Controller, error) {
+	store, err := newMongoStore(db)
+	if err != nil {
+		return Controller{}, err
+	}
+
+	return Controller{store: store}, nil
+
+}
+
+
+// GetAllFavorites gets all user favorite files by userID
+func (c Controller) GetAllFavorites(ctx context.Context, userID string) ([]string, error) {
+	filter := bson.D{
+		bson.E{
+			Key:   FavoriteBSONUserIDField,
+			Value: userID,
+		},
+	}
+
+	favoriteFiles, err := c.store.GetAll(ctx, filter)
+	if err != nil && err != mongo.ErrNoDocuments {
+		return nil, err
+	}
+
+	var returnedFavFiles []string
+
+	for _, fileob := range favoriteFiles {
+		returnedFavFiles = append(returnedFavFiles, fileob[2].Value.(string))
+	}
+
+	if err == mongo.ErrNoDocuments {
```

Comment:
Put this line before the 'var returnedFavFiles []string' line

Reply:
done